packages/contracts-bedrock/scripts/checks/interfaces/main.go

```diff
@@ -384,3 +470,93 @@ func getString(m map[string]interface{}, key string) string {
 	}
 	return ""
 }
+
+// Check if a source contract is in the exclude list
+func isExcludedSourceContract(contractName string) bool {
+	for _, excluded := range excludeSourceContracts {
+		if excluded == contractName {
+			return true
+		}
+	}
+	return false
+}
+
+// Function to verify that all contracts in the src directory have corresponding interfaces
+func verifyAllContractsHaveInterfaces() error {
+	// Get the current working directory
+	cwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory: %w", err)
+	}
+
+	var interfaceTracker = make(map[string]bool)
+
+	// First, get all interface files to track which interfaces exist
+	interfaceFiles, err := common.FindFiles([]string{"forge-artifacts/**/*.json"}, []string{})
```

Comment:
Interface files are always located at a predictable path, so we can simplify a lot of this code. For a given contract at `src/path/to/my/Contract.sol` you will always find the interface at `interfaces/path/to/my/IContract.sol`. For example, `src/L1/OptimismPortal2.sol` will have an interface at `interfaces/L1/IOptimismPortal2.sol`.

Using this fact should make it way simpler and avoid needing to search for the interface files, it also means that this script will enforce the correct path for the interface file which would be nice.